File: COMPILER.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"

int main(void) {
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
//  freopen("output.txt", "w", stdout);
#endif

    int t;
    cin >> t;
    while (t--) {
        string str;
        cin >> str;
        stack<char> st;
        int Count = 0;
        for (int i = 0; i < str.size(); i++) {
            if (str[i] == '<') {
                st.push(str[i]);
            } else {
                if (st.empty())
                    break;
                else
                    st.pop();
                if (st.empty())
                    Count = i + 1;
            }
        }
        cout << Count << endl;
    }
}

File: COMPILER.py
t = int(input())
while t:
	t -= 1
	s = input()
	st = list()
	Count = 0
	for i in range(len(s)):
		if s[i] == '<':
			st.append(s[i])
		else:
			if len(st) == 0:
				break
			else:
				st.pop()
			if len(st) == 0:
				Count = i + 1
	print(Count)
